packages/grid/_modules_/grid/components/columnSelection/GridHeaderCheckbox.tsx

```diff
@@ -52,15 +51,12 @@ const GridHeaderCheckbox = React.forwardRef<HTMLInputElement, GridColumnHeaderPa
     const isChecked = (totalSelectedRows > 0 && totalSelectedRows === totalRows) || isIndeterminate;
 
     const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-      const checked = event.target.checked;
+      const params: GridHeaderSelectionCheckboxParams = {
+        value: event.target.checked,
+        indeterminate: event.target.indeterminate,
```

Comment:
It's always false.

![image](https://user-images.githubusercontent.com/42154031/135558464-470dc592-0d73-4cb4-86f8-217e099c7859.png)

The `indeterminate` prop is not apply to the input: https://github.com/mui-org/material-ui/blob/3e679ac9e368aeb170d564d206d59913ceca7062/packages/mui-material/src/Checkbox/Checkbox.js#L79

Reply:
I copy pasted the old behavior.
But I can switch to `isIndeterminate`

Reply:
In https://github.com/mui-org/material-ui-x/issues/2672, if you have `shouldLimitSelectionToCurrentPage = false` and `props.pagination = true`, you are still using  `gridPaginatedVisibleSortedGridRowIdsSelector`

Reply:
There's a small regression here. With `checkboxSelectionVisibleOnly=false` it should display the header checkbox in an indeterminate state when going to the second page. Could you revert the part where you're only considering the rows in the current page when deciding whether is indeterminate or not? I forgot about this in https://github.com/mui-org/material-ui-x/issues/2672#issuecomment-928132290. 

Also, I guess we need to always pass true to the 3rd argument of `apiRef.current.selectRows` to unselect all rows in all pages when `checkboxSelectionVisibleOnly=false`. #2672 will fix the behavior when it's `true`.

![hCYScjeaBL](https://user-images.githubusercontent.com/42154031/135672158-a23711e9-3d71-4308-8bba-6fbef97ab457.gif)

Reply:
@m4theushw I think with my latest commit it works without the 3rd argument.
If you have a better name for `currentSelectionChunkSelectedCount`, my idea is to compute the list of the IDs that the checkbox can toggle and then only base the isChecked / isIndeterminate on this list.

Reply:
As discussed during the checkpoint meeting, let's keep the old behavior https://github.com/mui-org/material-ui-x/pull/2456#discussion_r720479373, even though it's not working as it should. Later, in #2672 it can be fixed.

Reply:
@m4theushw done